miden/src/cli/data.rs

```diff
@@ -406,6 +406,7 @@ impl ProgramFile {
         L: Library,
     {
         print!("Compiling program... ");
+        stdout().flush().expect("Couldn't flush stdout");
```

Comment:
Why is `.flush()` needed here?

Also, we should probably replace all `print!` and `println!` statements with `info!` (this might require some updates to how we structure messages as I'm not sure we can print on the same line with `info!`).

Reply:
I was using `.flush()` here to be sure that the message of the logger will be exactly in the desired place. I want to print this info between "Compiling program... " and "done" strings, but since `print!` doesn't flush the output stream, the logger message appears to be after "Compiling program... done" line. 

But it shouldn't be a problem after I change all `print`s to the `info`s, so I'll be able to remove this `flush`.